Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/java/org/apromore/plugin/portal/processdiscoverer/controllers/AnimationController.java

```diff
@@ -1,90 +1,96 @@
-        sb.append("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
-                "<definitions xmlns=\"http://www.omg.org/spec/BPMN/20100524/MODEL\"\n " +
-                "xmlns:dc=\"http://www.omg.org/spec/DD/20100524/DC\"\n " +
-                "xmlns:bpmndi=\"http://www.omg.org/spec/BPMN/20100524/DI\"\n " +
-                "xmlns:di=\"http://www.omg.org/spec/DD/20100524/DI\"\n " +
-                "xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\n " +
-                "targetNamespace=\"http://www.omg.org/bpmn20\"\n " +
-                "xsi:schemaLocation=\"http://www.omg.org/spec/BPMN/20100524/MODEL BPMN20.xsd\">");
-        sb.append(definitions.exportElements());
-        sb.append("</definitions>");
-        String bpmnText = sb.toString();
-        bpmnText.replaceAll("\n", "");
-        return bpmnText;
-    }
-}
+/*-
+ * #%L
+ * This file is part of "Apromore Core".
+ * %%
+ * Copyright (C) 2018 - 2020 Apromore Pty Ltd.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+
+package org.apromore.plugin.portal.processdiscoverer.controllers;
+
+import org.apromore.plugin.portal.processdiscoverer.PDController;
+import org.apromore.plugin.portal.processdiscoverer.vis.MissingLayoutException;
+import org.apromore.processdiscoverer.Abstraction;
+import org.apromore.processmining.models.graphbased.directed.bpmn.BPMNDiagram;
+import org.apromore.processmining.models.jgraph.ProMJGraph;
+import org.apromore.processmining.plugins.bpmn.BpmnDefinitions;
+import org.zkoss.zk.ui.event.Event;
+import org.zkoss.zul.Messagebox;
+
+public class AnimationController extends AbstractController {
+    public AnimationController(PDController controller) {
+        super(controller);
+    }
+    
+    @Override
+    public void onEvent(Event event) throws Exception {
+        if (!parent.prepareCriticalServices()) {
+            return;
+        }
+        
+        if (parent.getLogAnimationPluginCE() == null) {
+            Messagebox.show("Log Animation Plugin CE version is not available.");
+            return;
+        }
+        
+        Abstraction abs = parent.getOutputData().getAbstraction();
+        if (abs.getLayout() == null) {
+            throw new MissingLayoutException("Missing layout of the process map for animating");
+        }
+        
+        if (parent.getBPMNMode()) {

```

Comment:
it would be gr8 to move it to just 1 execute , with checks based on parameters, I see the execute method is almost the same.
But this is good if the intention is to delete the old execute method eventually

Reply:
Ok, will make sure to remove duplication when making changes to the other execute method (not yet updated).